From eb5a6181cfa451338adebccb69091a77ee767a5f Mon Sep 17 00:00:00 2001
From: ZENG Hao <c@cyano.cn>
Date: Tue, 2 Dec 2025 06:56:35 +0800
Subject: [PATCH 40/40] mingw: Fix Win32 UTF-8 manifest conformance

Microsoft's documentation for UTF-8 code page [1] gives a non-conforming
manifest. Windows's XP won't load exe files with this manifest.

The documentation for application manifests [2] provides the conforming
version. It is compatible with Windows XP.

[1] https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page#examples
[2] https://learn.microsoft.com/en-us/windows/win32/sbscs/application-manifests#activeCodePage

gcc/ChangeLog:

	* config/i386/winnt-utf8.manifest: Fix conformance.
---
 gcc/config/i386/winnt-utf8.manifest | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/gcc/config/i386/winnt-utf8.manifest b/gcc/config/i386/winnt-utf8.manifest
index dab929e1515..6869a3917f3 100644
--- a/gcc/config/i386/winnt-utf8.manifest
+++ b/gcc/config/i386/winnt-utf8.manifest
@@ -1,8 +1,9 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
-<assembly manifestVersion="1.0" xmlns="urn:schemas-microsoft-com:asm.v1">
-  <application>
-    <windowsSettings>
-      <activeCodePage xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">UTF-8</activeCodePage>
-    </windowsSettings>
-  </application>
+<assembly manifestVersion="1.0" xmlns="urn:schemas-microsoft-com:asm.v1"
+          xmlns:asmv3="urn:schemas-microsoft-com:asm.v3">
+  <asmv3:application>
+    <asmv3:windowsSettings xmlns="http://schemas.microsoft.com/SMI/2019/WindowsSettings">
+      <activeCodePage>UTF-8</activeCodePage>
+    </asmv3:windowsSettings>
+  </asmv3:application>
 </assembly>
-- 
2.54.0

